docs/how-to-guides.md

```diff
@@ -0,0 +1,383 @@
+directly convertible to make them work with Melange.
+
+For this reason, the recommendation is to migrate libraries at a time where they
+were compatible with past versions of ReScript, for example v9 or at least v10.
+
+These are the steps to follow:
+
+- Add an `opam` file
+- Add a `dune-project` file
+- Replace the `bsconfig.json` file with one or multiple `dune` files
+- (Optional) Migrate from ReScript syntax to Reason or OCaml syntaxes
+- Make sure everything works: `dune build @all`
+- Final step: remove `bsconfig.json` and adapt `package.json`
+
+Let's go through them in detail:
+
+### Add an `opam` file
+
+To migrate your ReScript library to Melange, you will need some packages.
+Melange is designed to be used with [opam](opam.ocaml.org/), the package manager
+of OCaml, which is explained in [its own section](./package-management.md).
+
+To get started with the library migration, let's create an `opam` file in your
+library's root folder with the minimum set of packages to start working:
+
+```text
+opam-version: "2.0"
+synopsis: "My Melange library"
+description: "A library for Melange"
+maintainer: ["<your_name>"]
+authors: ["<your_name>"]
+license: "XXX"
+homepage: "https://github.com/your/project"
+bug-reports: "https://github.com/your/project/issues"
+depends: [
+  "ocaml"
+  "dune"
+  "melange"
+]
+dev-repo: "git+https://github.com/your/project.git"
+```
+
+If your library was using [Reason syntax](https://reasonml.github.io/en/) (`re`
+files), you will need to add `"reason"` to the list of dependencies. If the
+library was using ReScript syntax (`res` files), you will need to add
+`rescript-syntax` to the list of dependencies. You can read more about how to
+migrate from ReScript syntax in the section below.
+
+At this point, we can create a [local opam
+switch](https://opam.ocaml.org/blog/opam-local-switches/) to start working on
+our library:
+
+```bash
+opam switch create . 5.1.0~rc2 -y --deps-only
+```
+
+Once this step is done, we can call `dune` from the library folder, but first we
+need some configuration files.
+
+### Add a `dune-project` file
+
+Create a file named `dune-project` in the library root folder. This file will
+tell Dune a few things about our project configuration:
+
+```text
+(lang dune 3.8)
+
+(using melange 0.1)
+```
+
+### Replace the `bsconfig.json` file with one or multiple `dune` files
+
+Now, we need to add a `dune` file where we will tell Dune about our library. You
+can put this new file next to the library sources, it will look something like
+this:
+
+```text
+(library
+ (name lib)
+ (modes melange))
+```
+
+Let's see how the most common configurations in `bsconfig.json` (or
+`rescript.json`) map to `dune` files. You can find more information about these
+configurations in the [Rescript
+docs](https://rescript-lang.org/docs/manual/latest/build-configuration) and in
+the [Dune docs](https://dune.readthedocs.io/en/stable/dune-files.html#library).
+
+#### `name`, `namespace`
```

Comment:
we should document that `"namespace": "foo-bar"` should translate to `(library (name fooBar))`

Reply:
https://github.com/melange-re/melange-re.github.io/pull/111/commits/e60d446cff11f95d6061577f40ec31165018e949